ksqldb-engine/src/main/java/io/confluent/ksql/KsqlExecutionContext.java

```diff
@@ -73,6 +74,8 @@
    */
   KsqlConfig getKsqlConfig();
 
+  MetricCollectors metricCollectors();
```

Comment:
This doesn't seem like a horrible expansion of the responsibility of the "execution contexct" to also expose the metrics registry. Adding this here allows us to save on adding the registry to the constructors of like 50 more classes.